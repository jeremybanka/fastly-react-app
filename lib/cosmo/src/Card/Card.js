/* @flow */

import * as React from "react";
import type { ComponentType } from "react";
import { FormattedMessage } from "react-intl";
import styled from "styled-components";
import { Box, Text } from "../";
import css from "@styled-system/css";

type TextType =
  | string
  | React.Element<typeof Text>
  | React.Element<typeof FormattedMessage>;

type CardProps = { children: React.Node };
const Card: ComponentType<CardProps> = styled("div")(
  css({
    display: "flex",
    flexDirection: "column",
    backgroundColor: "grays.0",
    // height: "100%", // problematic for unknown reasons
    // padding: "lg", // probably reinstate for sigsci, or control with prop
    // [`.SortableItem--dragging &,
    //   &:hover`]: { boxShadow: "lg" },
    // "& .card-header-buttons": { visibility: "hidden" },
    // [`&:hover .card-header-buttons,
    //   .SortableItem--dragging & .card-header-buttons`]: {
    //   visibility: "visible",
    // },
  })
);

type CardHeaderProps = { children: React.Node };
const CardHeader = (props: CardHeaderProps) => (
  <Box marginBottom="md">{props.children}</Box>
);

type CardTitleProps = { children: TextType };
const CardTitle = (props: CardTitleProps) => (
  <Box paddingBottom="sm" marginBottom="md" borderBottom="heavy">
    <Text fontWeight="bold">{props.children}</Text>
  </Box>
);

type CardSubtitleProps = { children: TextType };
const CardSubtitle = (props: CardSubtitleProps) => (
  <Box>
    <Text fontSize="sm" color="grays.4">
      {props.children}
    </Text>
  </Box>
);

type CardBodyProps = { children: React.Node };
const CardBody: ComponentType<CardBodyProps> = styled("div")({
  flex: "1 1 0%",
});

type CardFooterProps = { children: React.Node };
const CardFooter: ComponentType<CardFooterProps> = styled("div");

Card.Body = CardBody;
Card.Footer = CardFooter;
Card.Header = CardHeader;
Card.Title = CardTitle;
Card.Subtitle = CardSubtitle;

export default Card;
