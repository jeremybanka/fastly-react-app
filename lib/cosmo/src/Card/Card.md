```jsx
<Card>
  <Card.Header>
    <Card.Title>Queen of hearts</Card.Title>
    <Card.Subtitle>Not a bad draw</Card.Subtitle>
  </Card.Header>
  <Card.Body>
    Face cards tend to diverge from the normal appearance of cards with a value
    between 2 and 10
  </Card.Body>
</Card>
```
