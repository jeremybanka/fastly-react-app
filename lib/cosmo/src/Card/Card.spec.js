/* @flow */

import React from "react";
import { render } from "enzyme";
import Card from "./Card";

describe("Card test suite", () => {
  const card = (
    <Card>
      <Card.Header>
        <Card.Title>Queen of hearts</Card.Title>
        <Card.Subtitle>Not a bad draw</Card.Subtitle>
      </Card.Header>
      <Card.Body>
        Face cards tend to diverge from the normal appearance of cards with a
        value between 2 and 10
      </Card.Body>
    </Card>
  );
  it("renders the stored snapshot", () => {
    expect(render(card)).toMatchSnapshot();
  });
});
