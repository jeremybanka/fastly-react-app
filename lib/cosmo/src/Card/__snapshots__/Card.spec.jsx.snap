// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card test suite renders the stored snapshot 1`] = `
<div
  class="Flexbox-sc-18yd7vq-0 Card___StyledFlexbox-sc-1w5gsob-0 gxblUI"
>
  <div
    class="Box-sc-1v2etov-0 guzVPE"
  >
    <div
      class="Box-sc-1v2etov-0 kDqmRE"
    >
      <span
        class="Text-sc-6c1xd4-0 inZhph"
        font-weight="bold"
      >
        Queen of hearts
      </span>
    </div>
    <div
      class="Box-sc-1v2etov-0 kAhoak"
    >
      <span
        class="Text-sc-6c1xd4-0 NIkAq"
        color="grays.4"
        font-size="sm"
      >
        Not a bad draw
      </span>
    </div>
  </div>
  <div
    class="Box-sc-1v2etov-0 Card___StyledBox-sc-1w5gsob-1 lhVxmz"
  >
    Face cards tend to diverge from the normal appearance of cards with a value between 2 and 10
  </div>
</div>
`;
